scripts/github-actions/generate-module-index-data.js

```diff
@@ -98,13 +117,114 @@ async function generateModuleIndexData({ require, github, context, core }) {
+        .replace(/[\/\\]/g, "-");
+      const moduleBicepRegistryRef = moduleBicepRegistryRefReplace;
+
+      try {
+        core.info(`Processing AVM Resource Module ${modulePath}:...`);
+        core.info(`  Retrieving AVM Resource Module ${versionListUrl}`);
+        core.info(`    Git Tag: ${modulePath}`);
+        core.info(`    BRM Ref: ${moduleBicepRegistryRef}`);
+
+        const versionListResponse = await axios.get(versionListUrl);
+        const tags = versionListResponse.data.tags.sort();
+
+        const properties = {};
+        for (const tag of tags) {
+          var description = await getModuleDescription(
+            github,
+            core,
+            path,
+            modulePath,
+            (moduleRoot = "avm/res"),
+            tag,
+            context
+          );
+          if (description) {
+            properties[tag] = { description };
+            modulesWithDescriptions[modulePath] = true;
+          }
+        }
+
+        moduleIndexData.push({
+          moduleName: moduleBicepRegistryRef,
+          tags,
+          properties,
+        });
+      } catch (error) {
+        core.setFailed(error);
+      }
+    }
+  }
+
+  // AVM Pattern Modules
+  for (const moduleName of moduleGroupsAvmPtn) {
+    const modulePath = `avm/ptn/${moduleName}`;
+    const versionListUrl = `https://mcr.microsoft.com/v2/bicep/${modulePath}/tags/list`;
+    const moduleBicepRegistryRefSplit = modulePath
+      .split(/[\/\\]avm[\/\\]/)
+      .pop();
+    const moduleBicepRegistryRefReplace = moduleBicepRegistryRefSplit
+      .replace(/-/g, "")
+      .replace(/[\/\\]/g, "-");
+    const moduleBicepRegistryRef = moduleBicepRegistryRefReplace;
+
+    try {
+      core.info(`Processing AVM Pattern Module ${modulePath}:...`);
+      core.info(`  Retrieving AVM Pattern Module ${versionListUrl}`);
+      core.info(`    Git Tag: ${modulePath}`);
+      core.info(`    BRM Ref: ${moduleBicepRegistryRef}`);
+
+      const versionListResponse = await axios.get(versionListUrl);
+      const tags = versionListResponse.data.tags.sort();
+
+      const properties = {};
+      for (const tag of tags) {
+        var description = await getModuleDescription(
+          github,
+          core,
+          path,
+          modulePath,
+          (moduleRoot = "avm/ptn"),
+          tag,
+          context
+        );
+        if (description) {
+          properties[tag] = { description };
+          modulesWithDescriptions[modulePath] = true;
+        }
+      }
+
+      moduleIndexData.push({
+        moduleName: moduleBicepRegistryRef,
+        tags,
+        properties,
+      });
+    } catch (error) {
+      core.setFailed(error);
+    }
+  }
+
   core.info(`Writing moduleIndex.json`);
   await fs.writeFile(
     "moduleIndex.json",
     JSON.stringify(moduleIndexData, null, 2)
   );
 
-  core.info(`Processed ${moduleGroups.length} modules groups.`);
+  core.info(`Processed ${allModuleGroups.length} modules groups.`);
```

Comment:
It seems we only use `allModuleGroups` to log how many module groups are processed. I think we can just remove `allModuleGroups` and replace `allModulegroups.length` with `moduleGroups.length + moduleGroupsAvmRes.length + moduleGroupsAvmPtn.length`.

Reply:
changed